### PR TITLE
TPROD-400: Remove fetchAll Deprecation 

### PR DESCRIPTION
**Pull request by mabumusa1**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.

### CODE REVIEW

File: app/bundles/LeadBundle/Entity/LeadFieldRepository.php [hunk 1]
```diff
@@ -356,15 +356,16 @@ public function compareDateValue($lead, $field, $value)
         $q->select('l.id')
             ->from(MAUTIC_TABLE_PREFIX.'leads', 'l')
             ->where(
-                $q->expr()->andX(
+                $q->expr()->and(
```
Comment (mabumusa1) [inline app/bundles/LeadBundle/Entity/LeadFieldRepository.php:377]: @escopecz I made this on purpose, I read your comments regarding the replacement of `andX` with `and` on Jira but I do not think this has a huge impact

Comment (escopecz) [reply app/bundles/LeadBundle/Entity/LeadFieldRepository.php:377]: Thanks for pointing this out. It's OK to replace it in methods where we are certain that only the DBAL query builder is used.

**Comment by escopecz** [comment]
The changes look fine, but it seems there are 196 usages of the deprecated fetchAll() method. Do you plan to refactor that too?

**Comment by mabumusa1** [comment]
@escopecz yes, but this PR is looking into it PROD 400, after we get list in Jira merged we can look into phpunit deprecation list and resolve

**Comment by escopecz** [comment]
Well, it's not only about resolving the visible deprecations pointed out by the PHPUNIT tests. Mautic has a big test coverage deficit so it's won't show all usages. We should resolve them so the code works once we upgrade Doctrine. Do you want to create another ticket for the rest of these deprecations?

**Comment by mabumusa1** [comment]
> Well, it's not only about resolving the visible deprecations pointed out by the PHPUNIT tests. Mautic has a big test coverage deficit so it's won't show all usages. We should resolve them so the code works once we upgrade Doctrine. Do you want to create another ticket for the rest of these deprecations?

I see, no I will use this PR to resolve deprecations

**Comment by escopecz** [comment]
Another option is to check if this method still exists in the current Doctrine version and if it needs refactoring in all places. Maybe we can cut corners here and leave it for our future selves as it would not block the upgrade ;)

**Comment by mabumusa1** [comment]
> Another option is to check if this method still exists in the current Doctrine version and if it needs refactoring in all places. Maybe we can cut corners here and leave it for our future selves as it would not block the upgrade ;)

I resolved all the occurrences, I do not want to keep this for the future, let's get Mautic up to date so we can focus on features.

File: app/bundles/CoreBundle/Test/Doctrine/DBALMocker.php [hunk 2]
```diff
@@ -276,6 +277,10 @@ public function getMockResultStatement()
             ->method('fetch')
             ->willReturn($this->queryResponse);
 
+        $mock->expects($this->testCase->any())
```
Comment (mabumusa1) [inline app/bundles/CoreBundle/Test/Doctrine/DBALMocker.php:282]: @escopecz I added this to the mock, I know the method to does not exists in the original class and this mocker needs update

File: app/bundles/CoreBundle/Test/Doctrine/RepositoryConfiguratorTrait.php [hunk 3]
```diff
@@ -5,7 +5,7 @@
 namespace Mautic\CoreBundle\Test\Doctrine;
 
 use Doctrine\DBAL\Connection;
-use Doctrine\DBAL\Driver\ResultStatement;
```
Comment (mabumusa1) [inline app/bundles/CoreBundle/Test/Doctrine/RepositoryConfiguratorTrait.php:8]: @escopecz `ResultStatement` is not returned anymore

File: app/bundles/EmailBundle/EventListener/ReportSubscriber.php [hunk 4]
```diff
@@ -525,8 +525,8 @@ public function onReportGraphGenerate(ReportGraphEvent $event)
                     );
                     $this->addDNCTableForEmails($queryBuilder);
                     $queryBuilder->resetQueryPart('groupBy');
-                    $counts = $queryBuilder->execute()->fetch();
-                    $chart  = new PieChart();
+                    $counts   = $queryBuilder->execute()->fetchOne();
```
Comment (mabumusa1) [inline app/bundles/EmailBundle/EventListener/ReportSubscriber.php]: @escopecz just to verify that is the right method

File: app/bundles/EmailBundle/Tests/Stats/SentHelperTest.php [hunk 5]
```diff
@@ -6,7 +6,7 @@
 
 use Doctrine\DBAL\Connection;
 use Doctrine\DBAL\Query\QueryBuilder;
-use Doctrine\DBAL\Statement;
```
Comment (mabumusa1) [inline app/bundles/EmailBundle/Tests/Stats/SentHelperTest.php:9]: @escopecz `Statement` is not used anymore it is `Result`

**Comment by mabumusa1** [comment]
@escopecz this PR solves these depercation as well 

```
  4x: The Doctrine\DBAL\Driver\ResultStatement::fetchColumn method is deprecated (Use fetchOne() instead.).
    2x in ReferenceResolverTest::testResolveLeadReferences from Mautic\IntegrationsBundle\Tests\Unit\Sync\SyncDataExchange\Internal\Executioner
    1x in VersionProviderTest::testGetVersionForMySql from Mautic\CoreBundle\Tests\Unit\Doctrine\Provider
    1x in VersionProviderTest::testGetVersionForMariaDb from Mautic\CoreBundle\Tests\Unit\Doctrine\Provider


  3x: The Doctrine\DBAL\Statement::execute method is deprecated (Statement::execute() is deprecated, use Statement::executeQuery() or executeStatement() instead).
    1x in MaxMindDoNotSellPurgeCommandTest::testCommandDryRun from Mautic\CoreBundle\Tests\Unit\Command
    1x in MaxMindDoNotSellPurgeCommandTest::testNoContactsFound from Mautic\CoreBundle\Tests\Unit\Command
    1x in MaxMindDoNotSellPurgeCommandTest::testPurge from Mautic\CoreBundle\Tests\Unit\Command

  3x: The Doctrine\DBAL\Statement::fetchAllAssociative method is deprecated (Use Result::fetchAllAssociative() instead).
    1x in MaxMindDoNotSellPurgeCommandTest::testCommandDryRun from Mautic\CoreBundle\Tests\Unit\Command
    1x in MaxMindDoNotSellPurgeCommandTest::testNoContactsFound from Mautic\CoreBundle\Tests\Unit\Command
    1x in MaxMindDoNotSellPurgeCommandTest::testPurge from Mautic\CoreBundle\Tests\Unit\Command
```

**Comment by mabumusa1** [comment]
Note to self, in another PR remove references for `fetch` and remove all references for (if any) AFAIK it is only in `app/bundles/CoreBundle/Test/Doctrine/DBALMocker.php`

```
                'fetch',
                'fetchAll',
                'fetchOne',
```


There are 40 references for `fetch`, I need to look for each query and verify which function should be replaced with 
```
* @deprecated Use fetchNumeric(), fetchAssociative() or fetchOne() instead.
```

This is done via commit fc69b9d1a627ab6046feef0c8471c0c653ae6f5e

File: app/bundles/EmailBundle/EventListener/ReportSubscriber.php (hunk 4 above)
Comment (escopecz) [reply app/bundles/EmailBundle/EventListener/ReportSubscriber.php]: I think it should be https://www.doctrine-project.org/projects/doctrine-dbal/en/current/reference/data-retrieval-and-manipulation.html#fetchassociative based on how the `$counts` variable is used afterwards

File: app/bundles/LeadBundle/Entity/LeadDeviceRepository.php [hunk 6]
```diff
@@ -156,15 +156,14 @@ public function isAnyLeadDeviceTracked(Lead $lead)
...
-            ->setParameter('leadId', (int) $lead->getId())
-            ->orderBy('date_added', 'desc')
-            ->execute()
-            ->fetchAll();
+        ->from(MAUTIC_TABLE_PREFIX.'lead_devices', 'es')
+        ->where('lead_id = :leadId')
+        ->setParameter('leadId', (int) $lead->getId())
+        ->orderBy('date_added', 'desc')
+        ->execute()->fetchAllAssociative();
     }
```
Comment (escopecz) [inline app/bundles/LeadBundle/Entity/LeadDeviceRepository.php]: Most of the changes here are unnecessary and for the worse. Was there some reason for this style change?

File: app/bundles/LeadBundle/Entity/LeadFieldRepository.php [hunk 7]
```diff
@@ -372,15 +373,15 @@ public function compareDateValue($lead, $field, $value)
...
                     $q->expr()->eq('l.id', ':lead'),
                     $q->expr()->eq($property, ':value')
                 )
             )
             ->setParameter('lead', (int) $lead)
             ->setParameter('value', $value);
 
-        $result = $q->execute()->fetch();
+        $result = $q->execute()->fetchOne();
```
Comment (escopecz) [inline app/bundles/LeadBundle/Entity/LeadFieldRepository.php]:
```suggestion
        $result = $q->execute()->fetchAssociative();
```

File: app/migrations/Version20201102120710.php [hunk 8]
```diff
@@ -20,8 +20,8 @@ public function preUp(Schema $schema): void
 
         $sql  = 'SHOW INDEX FROM '.$this->table.' WHERE Key_name = "'.$this->index.'"';
         $stmt = $this->connection->prepare($sql);
-        $stmt->execute();
-        $indexExists = (bool) $stmt->fetch();
+        $stmt->executeStatement();
+        $indexExists = (bool) $stmt->fetchOne();
         $stmt->closeCursor();
```
Comment (escopecz) [inline app/migrations/Version20201102120710.php]: Isn't the `fetchOne` method also deprecated here? I solved it like this: https://github.com/mautic/mautic/pull/12217/files#diff-ec5e5e166db84f8d189cb1bf6af47c2bb19500ef7a70e5eafd4c833c94e65888R24

File: app/bundles/LeadBundle/Entity/LeadListRepository.php [hunk 9]
```diff
@@ -295,7 +295,7 @@ public function getLeadCount($listIds)
             ->setParameter('false', false, 'boolean')
             ->groupBy('l.leadlist_id');
 
-        $result = $q->execute()->fetchAll();
+        $result =  $q->execute()->fetchAllAssociative();
```
Comment (escopecz) [inline app/bundles/LeadBundle/Entity/LeadListRepository.php]: I see the new space added in many places. Why is that?

File: app/bundles/LeadBundle/Entity/LeadRepository.php [hunk 10]
```diff
@@ -1057,10 +1057,10 @@ public function updateLastActive($leadId)
...
     public function getMaxLeadId()
     {
-        $result = $this->getEntityManager()->getConnection()->createQueryBuilder()
-            ->select('max(id) as max_lead_id')
-            ->from(MAUTIC_TABLE_PREFIX.'leads', 'l')
-            ->execute()->fetchAll();
+        $result =         $this->getEntityManager()->getConnection()->createQueryBuilder()
+        ->select('max(id) as max_lead_id')
+        ->from(MAUTIC_TABLE_PREFIX.'leads', 'l')
+        ->execute()->fetchAllAssociative();
```
Comment (escopecz) [inline app/bundles/LeadBundle/Entity/LeadRepository.php]: The indentation is corrupted here too

File: app/bundles/LeadBundle/Segment/ContactSegmentService.php [hunk 11]
```diff
@@ -361,7 +361,7 @@ private function timedFetchAll(QueryBuilder $qb, $segmentId)
     {
         try {
             $start  = microtime(true);
-            $result = $qb->execute()->fetchAll(\PDO::FETCH_ASSOC);
+            $result = $qb->execute()->fetchAll();
```
Comment (escopecz) [inline app/bundles/LeadBundle/Segment/ContactSegmentService.php]: This is not replacing the deprecated method, right?

File: app/bundles/LeadBundle/Tests/Controller/LeadControllerTest.php [hunk 12]
```diff
@@ -569,18 +569,17 @@ private function createCampaign(): Campaign
...
-            ->where("cl.lead_id = {$leadId}")
-            ->orderBy('cl.company_id')
-            ->execute()
-            ->fetchAll();
+        ->select('cl.lead_id, cl.company_id, cl.is_primary, c.companyname')
+        ->from(MAUTIC_TABLE_PREFIX.'companies_leads', 'cl')
+        ->join('cl', MAUTIC_TABLE_PREFIX.'companies', 'c', 'c.id = cl.company_id')
+        ->where("cl.lead_id = {$leadId}")
+        ->orderBy('cl.company_id')
+        ->execute()->fetchAllAssociative();
```
Comment (escopecz) [inline app/bundles/LeadBundle/Tests/Controller/LeadControllerTest.php]: Please revert the indentation

File: app/bundles/NotificationBundle/Entity/StatRepository.php [hunk 13]
```diff
@@ -247,7 +247,8 @@ public function getSentCounts($notificationIds = [], \DateTime $fromDate = null)
         $q->groupBy('s.notification_id');
 
         //get a total number of sent notifications first
-        $results = $q->execute()->fetchAll();
+        $run     = $q->execute();
+        $results = $run->fetchAllAssociative();
```
Comment (escopecz) [inline app/bundles/NotificationBundle/Entity/StatRepository.php]: I see this in many places. It's nit-pick but variables should be a noun and not a verb. Can it be returned to the one-liner it was before? If possible in all places this was split.

File: app/bundles/PageBundle/Entity/HitRepository.php [hunk 14]
```diff
@@ -280,13 +280,12 @@ public function getBounces($pageIds, \DateTime $fromDate = null, $isVariantCheck
     {
         $inOrEq = (!is_array($pageIds)) ? 'eq' : 'in';
 
-        $hitsColumn = ($isVariantCheck) ? 'variant_hits' : 'unique_hits';
-        $q          = $this->getEntityManager()->getConnection()->createQueryBuilder();
-        $pages      = $q->select("p.id, p.$hitsColumn as totalHits, p.title")
+        $hitsColumn   = ($isVariantCheck) ? 'variant_hits' : 'unique_hits';
+        $q            = $this->getEntityManager()->getConnection()->createQueryBuilder();
+        $pages        = $q->select("p.id, p.$hitsColumn as totalHits, p.title")
```
Comment (escopecz) [inline app/bundles/PageBundle/Entity/HitRepository.php]: These are also unnecessary changes.

**Review by escopecz — CHANGES_REQUESTED** [review]

File: app/bundles/FormBundle/Entity/SubmissionRepository.php [hunk 15]
```diff
@@ -190,9 +191,10 @@ public function getEntity($id = 0)
                 ->from($this->getResultsTableName($form->getId(), $form->getAlias()), 'r')
                 ->where('r.submission_id = :id')
                 ->setParameter('id', $id);
-            $results = $q->execute()->fetchAll();
```
Comment (mabumusa1) [inline app/bundles/FormBundle/Entity/SubmissionRepository.php:193]: @escopecz please confirm that this change will not change the functionality

File: app/bundles/LeadBundle/Entity/LeadDeviceRepository.php (hunk 6 above)
Comment (mabumusa1) [reply app/bundles/LeadBundle/Entity/LeadDeviceRepository.php]: @escopecz I am not sure why `CSFixer` did not take care of that, I solved it anyways and all the other styling issues

File: app/bundles/LeadBundle/Entity/LeadFieldRepository.php (hunk 7 above)
Comment (mabumusa1) [reply app/bundles/LeadBundle/Entity/LeadFieldRepository.php]: Done

File: app/migrations/Version20201102120710.php (hunk 8 above)
Comment (mabumusa1) [reply app/migrations/Version20201102120710.php]: Thanks for the note, I took your code.

File: app/bundles/NotificationBundle/Entity/StatRepository.php (hunk 13 above)
Comment (mabumusa1) [reply app/bundles/NotificationBundle/Entity/StatRepository.php]: I fixed as it used to be before

File: app/bundles/LeadBundle/Segment/ContactSegmentService.php (hunk 11 above)
Comment (mabumusa1) [reply app/bundles/LeadBundle/Segment/ContactSegmentService.php]: Yes, but when I change it to `fetchAllAssociative`, `PHPStan` shows [an error ](https://github.com/mautic/mautic/actions/runs/4723025209/jobs/8378416839?pr=12218)

File: app/bundles/EmailBundle/EventListener/ReportSubscriber.php (hunk 4 above)
Comment (mabumusa1) [reply app/bundles/EmailBundle/EventListener/ReportSubscriber.php]: Fixed in commit 6aa36b3f377df06ce7bf7259f5dbfbc8d6403899

File: app/bundles/CoreBundle/Tests/Unit/Doctrine/Provider/VersionProviderTest.php [hunk 16]
```diff
@@ -5,21 +5,33 @@
...
+     */
     private $provider;
 
     protected function setUp(): void
     {
         parent::setUp();
 
         $this->connection           = $this->createMock(Connection::class);
-        $this->statement            = $this->createMock(Statement::class);
+        $this->result               = $this->createMock(Result::class);
```
Comment (dadarya0) [inline app/bundles/CoreBundle/Tests/Unit/Doctrine/Provider/VersionProviderTest.php:34]: too much spaces are here.

File: app/bundles/EmailBundle/Tests/Stats/SentHelperTest.php [hunk 17]
```diff
@@ -61,7 +61,7 @@ protected function setUp(): void
         $this->generatedColumnsProvider = $this->createMock(GeneratedColumnsProviderInterface::class);
         $this->userHelperMock           = $this->createMock(UserHelper::class);
         $this->queryBuilder             = $this->createMock(QueryBuilder::class);
-        $this->statement                = $this->createMock(Statement::class);
+        $this->result                   = $this->createMock(Result::class);
```
Comment (dadarya0) [inline app/bundles/EmailBundle/Tests/Stats/SentHelperTest.php:64]: too much spaces are here.

File: app/bundles/FormBundle/Entity/SubmissionRepository.php [hunk 18]
```diff
@@ -81,8 +82,8 @@ function ($value) {
         $this->buildWhereClause($dq, $args);
 
         //get a total count
-        $result = $dq->execute()->fetchAll();
-        $total  = $result[0]['count'];
+        $result  = $dq->execute()->fetchAllAssociative();
```
Comment (dadarya0) [inline app/bundles/FormBundle/Entity/SubmissionRepository.php:84]: Added new spaces hear also.

File: app/bundles/FormBundle/Entity/SubmissionRepository.php [hunk 19]
```diff
@@ -447,7 +450,7 @@ public function validateSubmissions($ids, $formId)
             );
 
         $validIds = [];
-        $results  = $q->execute()->fetchAll();
+        $results  =  $q->execute()->fetchAllAssociative();
```
Comment (dadarya0) [inline app/bundles/FormBundle/Entity/SubmissionRepository.php]:
```suggestion
        $results  = $q->execute()->fetchAllAssociative();
```

File: app/bundles/FormBundle/Model/SubmissionModel.php [hunk 20]
```diff
@@ -905,7 +905,7 @@ public function getTopSubmitters($limit = 10, $dateFrom = null, $dateTo = null,
         $chartQuery->applyFilters($q, $filters);
         $chartQuery->applyDateFilters($q, 'date_submitted');
 
-        return $q->execute()->fetchAll();
+        return $q->executeQuery()->fetchAllAssociative();
```
Comment (dadarya0) [inline app/bundles/FormBundle/Model/SubmissionModel.php]:
```suggestion
        return $q->execute()->fetchAllAssociative();
```
executeQuery is a method of Connection class and this method is not exist in QueryBuilder class.

File: app/bundles/IntegrationsBundle/Migration/AbstractMigration.php [hunk 21]
```diff
@@ -55,7 +55,7 @@ public function execute(): void
 
         foreach ($this->queries as $sql) {
             $stmt = $connection->prepare($sql);
-            $stmt->execute();
+            $stmt->executeStatement();
```
Comment (dadarya0) [inline app/bundles/IntegrationsBundle/Migration/AbstractMigration.php:58]: I believe it should be conditional as `executeQuery` should be used for select query and `executeStatement` should be used for Insert, Update and Delete queries.
what do you think @escopecz ?

File: app/bundles/LeadBundle/Entity/LeadDeviceRepository.php [hunk 22]
```diff
@@ -156,15 +156,15 @@ public function isAnyLeadDeviceTracked(Lead $lead)
      */
     public function getLeadDevices(Lead $lead)
     {
-        $qb = $this->getEntityManager()->getConnection()->createQueryBuilder();
+        $qb  = $this->getEntityManager()->getConnection()->createQueryBuilder();
```
Comment (dadarya0) [inline app/bundles/LeadBundle/Entity/LeadDeviceRepository.php]:
```suggestion
        $qb = $this->getEntityManager()->getConnection()->createQueryBuilder();
```

File: app/bundles/LeadBundle/Entity/LeadFieldRepository.php (hunk 7 above)
Comment (dadarya0) [reply app/bundles/LeadBundle/Entity/LeadFieldRepository.php]: still I can see fetchOne here instead fetchAssociative.

File: app/bundles/LeadBundle/Tests/Entity/LeadFieldRepositoryTest.php [hunk 23]
```diff
@@ -30,14 +31,16 @@ protected function setUp(): void
...
-        $builderCompare   = $this->createMock(QueryBuilder::class);
-        $statementAlias   = $this->createMock(Statement::class);
-        $statementCompare = $this->createMock(Statement::class);
-        $exprCompare      = $this->createMock(ExpressionBuilder::class);
+        $contactId                = 12;
+        $fieldAlias               = 'date_field';
+        $value                    = '2019-04-30';
+        $builderAlias             = $this->createMock(QueryBuilder::class);
+        $builderCompare           = $this->createMock(QueryBuilder::class);
+        $statementAlias           = $this->createMock(Statement::class);
```
Comment (dadarya0) [inline app/bundles/LeadBundle/Tests/Entity/LeadFieldRepositoryTest.php]: Mock of Result class is added then Mock of Statement class should be removed.
$statementAliasResul and $statementCompareResult should be used instead $statementAlias and $statementCompare.
Same need to be done in other places also in this file.
Please take care of extra spaces every where.

File: app/bundles/PageBundle/EventListener/ReportSubscriber.php [hunk 24]
```diff
@@ -524,8 +524,8 @@ public function onReportGraphGenerate(ReportGraphEvent $event)
                 case 'mautic.page.graph.pie.devices':
                     $queryBuilder->select('ds.device, COUNT(distinct(ph.id)) as the_count')
                         ->groupBy('ds.device');
-                    $data  = $queryBuilder->execute()->fetchAll();
-                    $chart = new PieChart();
+                    $data     = $queryBuilder->execute()->fetchAllAssociative();
```
Comment (dadarya0) [inline app/bundles/PageBundle/EventListener/ReportSubscriber.php:527]: Remove extra spaces here.

File: app/migrations/Version20201102120710.php [hunk 25]
```diff
@@ -18,10 +18,9 @@ public function preUp(Schema $schema): void
         $this->table = $this->getTableName();
         $this->index = $this->generatePropertyName($this->table, 'idx', ['email_id']);
 
-        $sql  = 'SHOW INDEX FROM '.$this->table.' WHERE Key_name = "'.$this->index.'"';
-        $stmt = $this->connection->prepare($sql);
-        $stmt->execute();
-        $indexExists = (bool) $stmt->fetch();
+        $sql         = 'SHOW INDEX FROM '.$this->table.' WHERE Key_name = "'.$this->index.'"';
+        $stmt        = $this->connection->prepare($sql);
+        $indexExists = (bool) $stmt->executeQuery()->fetchAllAssociative();
```
Comment (dadarya0) [inline app/migrations/Version20201102120710.php]:
```suggestion
        $indexExists = (bool) $stmt->executeQuery()->fetchAssociative();
```

**Review by dadarya0 — CHANGES_REQUESTED** [review]
@mabumusa1 I found some issues please check it.

File: app/bundles/CoreBundle/Tests/Unit/Doctrine/Provider/VersionProviderTest.php (hunk 16 above)
Comment (mabumusa1) [reply app/bundles/CoreBundle/Tests/Unit/Doctrine/Provider/VersionProviderTest.php:34]: Fixed in commit 1387c7eb8479a51b9db212aa5ead281c9b9c7164

File: app/bundles/EmailBundle/Tests/Stats/SentHelperTest.php (hunk 17 above)
Comment (mabumusa1) [reply app/bundles/EmailBundle/Tests/Stats/SentHelperTest.php:64]: Fixed in commit 1387c7eb8479a51b9db212aa5ead281c9b9c7164

File: app/bundles/FormBundle/Entity/SubmissionRepository.php (hunk 18 above)
Comment (mabumusa1) [reply app/bundles/FormBundle/Entity/SubmissionRepository.php:84]: Fixed in commit 1387c7eb8479a51b9db212aa5ead281c9b9c7164

File: app/bundles/FormBundle/Entity/SubmissionRepository.php (hunk 19 above)
Comment (mabumusa1) [reply app/bundles/FormBundle/Entity/SubmissionRepository.php]: Fixed in commit 1387c7eb8479a51b9db212aa5ead281c9b9c7164

File: app/bundles/LeadBundle/Entity/LeadDeviceRepository.php (hunk 22 above)
Comment (mabumusa1) [reply app/bundles/LeadBundle/Entity/LeadDeviceRepository.php]: Fixed in commit 1387c7eb8479a51b9db212aa5ead281c9b9c7164

File: app/bundles/PageBundle/EventListener/ReportSubscriber.php (hunk 24 above)
Comment (mabumusa1) [reply app/bundles/PageBundle/EventListener/ReportSubscriber.php:527]: Fixed in commit 1387c7eb8479a51b9db212aa5ead281c9b9c7164

**Comment by mabumusa1** [comment]
> @mabumusa1 I found some issues please check it.

Thanks @dadarya0, I am taking care of them. it would be  great if we can get this across the finish line today

File: app/bundles/IntegrationsBundle/Migration/AbstractMigration.php (hunk 21 above)
Comment (mabumusa1) [reply app/bundles/IntegrationsBundle/Migration/AbstractMigration.php:58]: @dadarya0 this function currently is used to execute migrations which usually does not contain queries, if we want this function to enable the ability to return records then your suggestion is valid.

File: app/bundles/LeadBundle/Tests/Entity/LeadFieldRepositoryTest.php (hunk 23 above)
Comment (mabumusa1) [reply app/bundles/LeadBundle/Tests/Entity/LeadFieldRepositoryTest.php]: Fixed in commit 3eb163cc33db937f70e871a4d07ebc143fd10bb3

File: app/bundles/FormBundle/Model/SubmissionModel.php (hunk 20 above)
Comment (mabumusa1) [reply app/bundles/FormBundle/Model/SubmissionModel.php]: fixed in commit 3eb163cc33db937f70e871a4d07ebc143fd10bb3

File: app/bundles/LeadBundle/Segment/ContactSegmentService.php (hunk 11 above)
Comment (mabumusa1) [reply app/bundles/LeadBundle/Segment/ContactSegmentService.php]: I made this change commit 2dee0c0d9de7f873b7c954f4fe553f5afa7c9704, to update the return values. yet we still need to tell phpstan that this line always get `Statement` in the return value not `int|string`

File: app/bundles/FormBundle/Entity/SubmissionRepository.php (hunk 15 above)
Comment (escopecz) [reply app/bundles/FormBundle/Entity/SubmissionRepository.php:193]: It looks correct to me.

File: app/bundles/LeadBundle/Entity/LeadDeviceRepository.php (hunk 6 above)
Comment (escopecz) [reply app/bundles/LeadBundle/Entity/LeadDeviceRepository.php]: Thanks!

File: app/bundles/LeadBundle/Entity/LeadFieldRepository.php (hunk 7 above)
Comment (mabumusa1) [reply app/bundles/LeadBundle/Entity/LeadFieldRepository.php]: Fixed in commit 550cd763cf203666e210066fe6bfefb95c037992

File: app/bundles/LeadBundle/Segment/ContactSegmentService.php (hunk 11 above)
Comment (mabumusa1) [reply app/bundles/LeadBundle/Segment/ContactSegmentService.php]: @escopecz  I need your help with this one, I fixed the function in `app/bundles/LeadBundle/Segment/Query/QueryBuilder.php` as you can see in commit 
550cd763cf203666e210066fe6bfefb95c037992 but `phpstan` is still showing issues, how do you tell it that `executeQuery` return `DriverStatement|DriverResultStatement` and `executeStatment` return `int|string`

Comment (escopecz) [reply app/bundles/LeadBundle/Segment/ContactSegmentService.php]: I see. This one is tricky. I looked around and it seem to me that the cleanest way how to fix this is to add this new method to the Segment QueryBuilder:
```php
public function fetchAllAssociative(): array
    {
        return $this->connection->fetchAllAssociative($this->getSQL(), $this->params, $this->paramTypes);
    }
```

And then use it here:
```php
$result = $qb->fetchAllAssociative();
```
WDYT?

File: app/bundles/IntegrationsBundle/Migration/AbstractMigration.php (hunk 21 above)
Comment (escopecz) [reply app/bundles/IntegrationsBundle/Migration/AbstractMigration.php:58]: I think that the queries in the `up` methods will be only DDL queries. See https://github.com/acquia/mc-cs-plugin-custom-objects/tree/staging/Migrations for examples. So I think that Mohammad's change is correct.

File: app/bundles/LeadBundle/Segment/ContactSegmentService.php (hunk 11 above)
Comment (mabumusa1) [reply app/bundles/LeadBundle/Segment/ContactSegmentService.php]: @escopecz taking a closer look into the code, the code we have 

```
    /**
     * Executes this query using the bound parameters and their types.
     *
     * Uses {@see Connection::executeQuery} for select statements and {@see Connection::executeUpdate}
     * for insert, update and delete statements.
     *
     * @return Result|int|string
     *
     * @throws \Doctrine\DBAL\DBALException
     */
    public function execute()
    {
        if (self::SELECT == $this->type) {
            return $this->connection->executeQuery($this->getSQL(), $this->params, $this->paramTypes);
        } else {
            return $this->connection->executeStatement($this->getSQL(), $this->params, $this->paramTypes);
        }
    }
```

Is just an old copy of `Doctrine\DBAL\Query\QueryBuilder` has

```
    /**
     * Executes this query using the bound parameters and their types.
     *
     * @return ForwardCompatibility\Result|int|string
     *
     * @throws Exception
     */
    public function execute()
    {
        if ($this->type === self::SELECT) {
            return ForwardCompatibility\Result::ensure(
                $this->connection->executeQuery($this->getSQL(), $this->params, $this->paramTypes)
            );
        }

        return $this->connection->executeStatement($this->getSQL(), $this->params, $this->paramTypes);
    }
```

I just removed our `execute` method and everything is working as expected

Comment (mabumusa1) [reply app/bundles/LeadBundle/Segment/ContactSegmentService.php]: Solved in commit 06b11f80bec21880218bdbc7b436020d67e22c6c

Comment (escopecz) [reply app/bundles/LeadBundle/Segment/ContactSegmentService.php]: Wow, that's interesting. Great solution!

File: app/bundles/LeadBundle/Segment/Query/QueryBuilder.php [hunk 26]
```diff
@@ -200,20 +201,19 @@ public function getState()
...
-     * @throws \Doctrine\DBAL\DBALException
+     * @throws \Exception
      */
     public function execute()
     {
-        if (self::SELECT == $this->type) {
-            return $this->connection->executeQuery($this->getSQL(), $this->params, $this->paramTypes);
-        } else {
-            return $this->connection->executeUpdate($this->getSQL(), $this->params, $this->paramTypes);
+        if (self::SELECT === $this->type) {
```
Comment (mabumusa1) [inline app/bundles/LeadBundle/Segment/Query/QueryBuilder.php:210]: @escopecz I had to restore this function but I am using the latest version of the code, the cause of this action is `$this->params, $this->paramTypes` are private properties in this class and they are not visible to the parent. I think this file needs restructuring

**Review by dadarya0 — APPROVED** [review]

**Review by escopecz — APPROVED** [review]
Thanks @mabumusa1 ! I see no more issues with the code changes 👍
